packages/rancher-gatekeeper/generated-changes/patch/values.yaml.patch

```diff
@@ -7,31 +7,31 @@
  auditMatchKindOnly: false
  constraintViolationsLimit: 20
  auditFromCache: false
-@@ -19,14 +19,17 @@
+@@ -22,14 +22,17 @@
    labelNamespace:
      enabled: true
      image:
--      repository: line/kubectl-kustomize
--      tag: 1.20.4-4.0.5
+-      repository: openpolicyagent/gatekeeper-crds
+-      tag: v3.7.1
 +      repository: rancher/kubectl
 +      tag: v1.20.2
        pullPolicy: IfNotPresent
        pullSecrets: []
 -image:
 -  repository: openpolicyagent/gatekeeper
 -  crdRepository: openpolicyagent/gatekeeper-crds
--  release: v3.6.0
+-  release: v3.7.1
```

Comment:
this is a breaking change, we should have logic for a user to fail a helm upgrade if these values are set or backwards compatibility with these old fields. Between the two, I'd prefer the former

Reply:
I've added the logic to fail the helm upgrade if the field `image` is overridden. Can you please review it once?

Reply:
Gatekeeper introduced the CRD image and the `crdRepository` logic with version 3.5.1, but this logic was never released in any of our rancher-gatekeeper charts because it doesn't follow the pattern the airgap exporter needs and therefore the images were not being picked up. It was quickly replaced by the new logic in this PR and it's what we've been using since then so it is not a breaking change. This also doesn't affect 2.5 because the latest gatekeeper chart in 2.5 is older than 3.5.1 so it doesn't have a CRD image.

Reply:
Sounds good, thanks for looking into this @PennyScissors. If it doesn't affect 2.5 and it was never released in a chart, I agree that validation is not necessary.

Reply:
@doflamingo721 can you revert the validation you added?

Reply:
@PennyScissors I reverted the validations, can you please review it again?